src/hdmf/common/resources.py

```diff
@@ -29,30 +29,53 @@ class Key(Row):
 
 class ResourceTable(Table):
     """
-    A table for storing the external resources a key refers to
+    A table for storing names of ontology sources and their uri
     """
 
     __defaultname__ = 'resources'
 
+    __columns__ = (
+        {'name': 'name', 'type': str,
+         'doc': 'The resource/registry that the term/symbol comes from.'},
+        {'name': 'resource_uri', 'type': str,
+         'doc': 'The URI for the resource term / registry symbol.'},
+    )
+
+
+class Resource(Row):
+    """
+    A Row class for representing rows in the ResourceTable
+    """
+
+    __table__ = ResourceTable
+
+
+class EntityTable(Table):
+    """
+    A table for storing the external resources a key refers to
+    """
+
+    __defaultname__ = 'entities'
+
     __columns__ = (
         {'name': 'keytable_idx', 'type': (int, Key),
          'doc': ('The index into the keys table for the user key that '
                  'maps to the resource term / registry symbol.')},
-        {'name': 'resource_name', 'type': str,
-         'doc': 'The resource/registry that the term/symbol comes from.'},
-        {'name': 'resource_entity_id', 'type': str,
+        {'name': 'resource_table_idx', 'type': (int, Resource),
```

Comment:
We should be consistent with line 61. Either have an underscore before "table" or not.

Reply:
This should also be changed in the schema. Sorry I didn't catch this inconsistency there.

Reply:
I personally prefer an underscore before "table" which means this needs to be changed throughout the code and schema.